tiledb/sm/query/query.cc

```diff
@@ -1181,14 +1181,26 @@ Status Query::check_buffer_names() {
 }
 
 Status Query::create_strategy(bool skip_checks_serialization) {
+  strategy_ = which_strategy(skip_checks_serialization);
+
+  if (strategy_ == nullptr) {
+    return logger_->status(
+        Status_QueryError("Cannot create strategy; allocation failed"));
+  }
+
+  return Status::Ok();
+}
+
+tdb_unique_ptr<IQueryStrategy> Query::which_strategy(
```

Comment:
This isn't a bad way to do this but I'm thinking it might cause issues. More precisely, if we have any of the constructors for the strategy do extensive work, we could be doing more work than what the user really needs. I think a better way might be to have this function return an ENUM value which would be sufficient for the query plan to know which strategy is used (a simple enum value to string function would be used) and create_strategy would just be a switch case creating the strategy according to the enum value. Thoughts?